Automattic/wp-calypso
# Help: Provide Link to User's Site in Forum Post

#### Changes proposed in this Pull Request

* When a user submits a post to the Forums, ensure that their URL is included. It also achieves the broader request made by @supernovia in https://github.com/Automattic/wp-calypso/issues/38280#issuecomment-567108997

#### Testing instructions

The way this is supposed to work is based on the number of sites which the user owns. You can compare the flows below:

| No Site                                                                                                                                                               | One site                                                                                                                                                              | Multiple sites                                                                                                                                                        |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| <img width="768" alt="Screenshot 2021-05-18 at 19 43 20" src="https://user-images.githubusercontent.com/43215253/118706486-4fefc300-b811-11eb-87d1-b7c2171da3f7.png"> | <img width="754" alt="Screenshot 2021-05-18 at 19 14 38" src="https://user-images.githubusercontent.com/43215253/118706253-1028db80-b811-11eb-963e-9bbf5f1652a3.png"> | <img width="752" alt="Screenshot 2021-05-18 at 19 35 52" src="https://user-images.githubusercontent.com/43215253/118706252-1028db80-b811-11eb-8308-b7050b46792a.png"> |

When a user enters the URL, it also performs some quick checks:

- If the site is a WordPress.com one owned by another account, it provides a link to the Account Recovery page since it can't be dealt with in the forums.
- If the site is not a WordPress.com one, and it's not connected with Jetpack, it provides a warning of that.

| <img width="751" alt="Screenshot 2021-05-20 at 19 29 57" src="https://user-images.githubusercontent.com/43215253/119030875-5eb9af80-b9a2-11eb-9602-c1e12b47a63b.png"> | <img width="756" alt="Screenshot 2021-05-20 at 19 29 43" src="https://user-images.githubusercontent.com/43215253/119030871-5bbebf00-b9a2-11eb-9d19-8db1f7de4986.png"> |
|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|

Depending on the options, one of the following messages will be included at the end of the Forums post:

- I don't have a site linked to this WordPress.com account
- I don't have a site with WordPress.com yet (different to the term above due to topics relating to Account Recover)
- The site I need help with is x.
- The site I need help with is x. This site is not linked to my WordPress.com account.

**Note:** If the site is self-hosted, a message either saying "It is not hosted by WordPress.com, but it is connected with Jetpack" or "It is not hosted by WordPress.com or connected with Jetpack" will be included too. This should save volunteers from having to manually check each blog, which can quickly become tiresome.

This also introduces an option to hide the URL in the Forums by setting it as an email; it'll automatically be hidden in the Forums due to this. You can see these changes in action below: https://wordpress.com/forums/topic/testing-please-ignore-apologies/

<img width="834" alt="Screenshot 2021-05-20 at 19 43 49" src="https://user-images.githubusercontent.com/43215253/119032131-bc9ac700-b9a3-11eb-904c-16a2c3f06399.png">

@supernovia @kokkieh I'd be really grateful for any feedback which you could provide about this. :) 

cc @jsnajdr 

Fixes #38280 
Fixes #16015
Fixes #16016
